baseplate.go

```diff
@@ -38,11 +39,23 @@ type Config struct {
 
 	Log     log.Config       `yaml:"log"`
 	Metrics metricsbp.Config `yaml:"metrics"`
+	Runtime runtimebp.Config `yaml:"runtime"`
 	Secrets secrets.Config   `yaml:"secrets"`
 	Sentry  log.SentryConfig `yaml:"setry"`
 	Tracing tracing.Config   `yaml:"tracing"`
 }
 
+// Args is the argument struct for baseplate.New
+type Args struct {
```

Comment:
Maybe just have `New` accept three arguments, context, path, and the config. I feel like this wrapping makes it harder to follow, lookup what needs to be done, etc. Unless we already know that there will be more things to add.

Reply:
Since go doesn't allow default values to args, using struct is a way to support that. Using 3 args directly means that even if the user don't have any additional service confg, they always need to explicitly pass `nil` into `New()`, while using `Args` struct means that it can be omitted completely.

Reply:
I'm not opposed to that, I mostly did this as a way to make `ServiceCfg` optional but that's not a huge benefit for this.